src/components/DropdownMenu.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+
+type Props = {
+  setSelectedPerson: (person: Person | null) => void;
+};
+
+const DELAY = 1000;
+
+export const DropdownMenu: React.FC<Props> = ({ setSelectedPerson }) => {
+  const [isMenuShown, setIsMenuShown] = useState(false);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const filteredPeople = useMemo(
+    () => peopleFromServer.filter((person: Person) => person.name
+      .toLowerCase().includes(appliedQuery.toLowerCase().trim())),
+    [appliedQuery],
+  );
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, DELAY), []);
+
+  const handleDropdownChange = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery('');
+    setAppliedQuery('');
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
```

Comment:
Need to close the list of users when `handleQueryChange` is triggered acoording to the requirements
![7621](https://github.com/mate-academy/react_autocomplete/assets/35075922/7eff8e9f-2d4b-4a4c-ba1f-b6dd826d16ea)

Reply:
Thanks for your comments!